### PR TITLE
Fix error caused by shared cookies between different app versions

### PR DESCRIPTION
**Pull request by ghost**
# Description

- fix https://github.com/ubiregiinc/stockscan-server/issues/1351

### CODE REVIEW

File: flaskext/kvsession.py [hunk 1]
```diff
@@ -137,9 +137,16 @@ def open_session(self, app, request):
...
-                        current_app.kvsession_store.get(sid_s))
-                    )
+                    try:
+                        s = KVSession(self.serialization_method.loads(
+                            current_app.kvsession_store.get(sid_s))
+                        )
+                    except (UnicodeDecodeError, ValueError):
+                        # If a cookie is reused between different app versions (Python 2 and Python 3), one of the
+                        # above exceptions will be raised. Just delete the associated session and create a new one
+                        current_app.kvsession_store.delete(sid_s)
```
Comment (pomu0325) [inline flaskext/kvsession.py:162]: DON’T delete sessions, but restore session from python2 serialization format and convert to python3 serialization format.

**Review by pomu0325 — CHANGES_REQUESTED** [review]

File: flaskext/kvsession.py (hunk 1 above)
Comment (pomu0325) [reply flaskext/kvsession.py:162]: another option: always use pickle protocol version 2

Comment (ghost) [reply flaskext/kvsession.py:162]: Yesterday, I first took the approach you mentioned above and found it worked. However, I encountered another serialisation error when downgrading the Python version from 3 to 2. I mean, the error occurred when unpickling an object which was pickled from Python 3 previously.  I'm not sure that we could fix the error in this case. So, I decided to take the DELETE-SESSION approach.

Comment (ghost) [reply flaskext/kvsession.py:162]: >Use protocol 2

Okay. Let me try.

Comment (ghost) [reply flaskext/kvsession.py:162]: I just updated the code. Please check it out.

Scenarios I have checked:
- Reusing a session which was created from Python 2 app version after upgrading to Python 3
- Reusing a session which was created from Python 3 app version (serialised with protocol 2) after downgrading to Python 2
- Reusing a session which was created from Python 3 app version (serialised with protocol 4) after downgrading to Python 2 => I'd like to keep deleting sessions if there is any error occurring.

**Pull request closed without merging**